lxd/endpoints/network_test.go

```diff
@@ -88,3 +90,20 @@ func newTCPListener(t *testing.T) *net.TCPListener {
 	require.NoError(t, err)
 	return listener
 }
+
+// Create IPv4 0.0.0.0 listener using random port
+// and ensure it is not accessible via IPv6 request.
+func TestEndpoints_NetworkCreateTCPSocketIPv4(t *testing.T) {
+	endpoints, config, cleanup := newEndpoints(t)
+	defer cleanup()
+
+	config.NetworkAddress = "0.0.0.0:0"
+	require.NoError(t, endpoints.Up(config))
+
+	address, certificate := endpoints.NetworkAddressAndCert()
+	parts := strings.Split(address, ":")
+	ipv6Address := fmt.Sprintf("[::1]:%s", parts[1])
+	message := fmt.Sprintf("Get https://%s/: dial tcp %s: connect: connection refused", ipv6Address, ipv6Address)
+
+	assert.EqualError(t, httpGetOverTLSSocket(ipv6Address, certificate), message)
```

Comment:
@dizlv should we also test that we can indeed connect to the IPv4 socket here, to prove its only listening on IPv4 (and not just not listening on anything).?